llvm/lib/CodeGen/GlobalISel/InlineAsmLowering.cpp

```diff
@@ -231,6 +232,16 @@ bool InlineAsmLowering::lowerInlineAsm(
   TargetLowering::AsmOperandInfoVector TargetConstraints =
       TLI->ParseConstraints(DL, TRI, Call);
 
+  const auto ConstraintError = [&](const GISelAsmOperandInfo &Info, Twine Msg) {
+    LLVMContext &Ctx = MIRBuilder.getContext();
+    Ctx.diagnose(DiagnosticInfoInlineAsm(
+        Call, "invalid constraint '" + Info.ConstraintCode + "' in '" +
+                  MF.getName() + "': " + Msg));
```

Comment:
Avoid using getName, it's broken for anonymous functions. The context should be reported by the caller based on the context instructions, you shouldn't need to include it here